python/ray/data/_internal/arrow_block.py

```diff
@@ -546,8 +577,12 @@ def aggregate_combined_blocks(
         """
 
         stats = BlockExecStats.builder()
+
+        keys = key if isinstance(key, list) else [key]
         key_fn = (
-            (lambda r: r[r._row.schema.names[0]]) if key is not None else (lambda r: 0)
+            (lambda r: tuple(r[r._row.schema.names[: len(keys)]]))
```

Comment:
Hi @LeonLuttenberger - taking a look at this code here. What's the reason we index the names by len(keys), instead of actually using the actual underlying key names?

ie - 
r[list(keys)]